In [73]:
import pandas as pd

In [ ]:
# df1 = pd.read_csv('D:\\ML CODES\\IBM\\Test_Cleaned\\Apple_test_cleaned.csv', encoding='latin-1')

In [ ]:
# df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3725 entries, 0 to 3724
Data columns (total 3 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   sentiment             3725 non-null   int64  
 1   sentiment:confidence  3725 non-null   float64
 2   text                  3725 non-null   object 
dtypes: float64(1), int64(1), object(1)
memory usage: 87.4+ KB


In [76]:
# df1 = df1.drop(['_golden','_unit_state','_trusted_judgments','_last_judgment_at','date','id','query','sentiment_gold'], axis=1)

In [77]:
# df1 = df1.to_csv("Apple_test_cleaned.csv", index=False)

In [78]:
# df1.isnull().sum()

##### The 82 rows with not_relevant sentiment have been removed. The DataFrame now only contains the relevant sentiment values:



In [79]:
# df1 = df1[df1['sentiment'] != 'not_relevant']
# df1['sentiment'].value_counts()

##### 79 rows with fewer than 5 words in the text column were removed (3804 → 3725 rows remaining). 

In [80]:
# print("Before:", len(df1))
# df1 = df1[df1['text'].str.split().str.len() >= 5]
# print("After:", len(df1))

In [81]:
# df1.to_csv("Apple_test_cleaned2.csv", index=False)

In [ ]:
# df1 = pd.read_csv("D:\\ML CODES\\IBM\\Test_Cleaned\\hotel_test_cleaned.csv")

In [ ]:
# df2.head()

,review,sentiment
0,One of the other reviewers has mentioned that ...,positive
1,A wonderful little production. <br /><br />The...,positive
2,I thought this was a wonderful way to spend ti...,positive
3,Basically there's a family where a little boy ...,negative
4,"Petter Mattei's ""Love in the Time of Money"" is...",positive


In [ ]:
# df2 = df2[['reviews.text']]

In [ ]:
# df2.to_csv('hotel_test_cleaned.csv',index=False)

In [ ]:
# df3 = pd.read_csv("D:\\ML CODES\\IBM\\test\\socialmedia-test.csv",encoding='latin-1')

In [271]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14640 entries, 0 to 14639
Data columns (total 2 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   airline_sentiment  14640 non-null  object
 1   text               14640 non-null  object
dtypes: object(2)
memory usage: 228.9+ KB


In [ ]:
# df3 = df3[['text']]

In [300]:
df1.head()

,Sentiment,Text
0,Positive,Enjoying a beautiful day at the park
1,Negative,Traffic was terrible this morning
2,Positive,Just finished an amazing workout
3,Positive,Excited about the upcoming weekend getaway
4,Neutral,Trying out a new recipe for dinner tonight


In [293]:
print("df2 duplicates:", df1.duplicated().sum())


df2 duplicates: 0


In [291]:
df1.drop_duplicates(inplace=True)

In [288]:
df1 = df1[['Sentiment','Text']]

In [292]:
df1.isnull().sum()

Sentiment    0
Text         0
dtype: int64

In [ ]:
# df1 = df1.drop([''],axis = 1)

##### Remove URLs

In [295]:
df1['Text'] = df1['Text'].str.replace(r"http\S+|www\S+", "", regex=True)

##### Remoev @ mentions


In [296]:
import re

df1["Text"] = df1["Text"].apply(lambda x: re.sub(r"@\w+", "", x))

##### Remove Punctuations

In [297]:
df1["Text"] = df1["Text"].apply(lambda x: re.sub(r"[^\w\s]", "", x))

#### handle elongated words

In [298]:


def remove_elongated_words(text):
    return re.sub(r'(.)\1{2,}', r'\1\1', text)

df1["Text"] = df1["Text"].apply(remove_elongated_words)

#### REMOVE NO_ASCII SPECIAL CHARACTER THAT CREATED DUE TO ENCOING


In [299]:
# Remove non-ASCII special characters (Ã, Â, etc.)
df1['Text'] = df1['Text'].str.replace(r'[^\x00-\x7F]+', '', regex=True)


#### SAVING FILE BACK 

In [301]:
df1.to_csv("Sentiment_train_cleaned.csv", index=False)

In [ ]:
# df1 = pd.read_csv('D:\\ML CODES\\IBM\\test\\test.csv')

In [ ]:
# df1 = pd.read_csv('D:\\ML CODES\\IBM\\test\\testdata.manual.tweet.csv',encoding='latin-1')

#### WROKING WITH TRAIN DATA

In [286]:
df1 = pd.read_csv('D:\\ML CODES\\IBM\\train\\sentiment train.csv',encoding= 'latin-1')